vcd/auth_saml_test.go

```diff
@@ -0,0 +1,101 @@
+// +build auth ALL functional
+
+package vcd
+
+import (
+	"os"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+)
+
+// TestAccVcdSamlAuth explicitly tests a simple operation using SAML auth when explicit SAML
+// testing user and password are specified.
+// Note. The test cannot be run in parallel because it temporarily overrides authentication cache
+// and credentials for the purpose of its run. It restores them at the end.
+func TestAccVcdSamlAuth(t *testing.T) {
+
+	// Skip test if explicit SAML credentials are not specified
+	if testConfig.Provider.SamlUser == "" || testConfig.Provider.SamlPassword == "" {
+		t.Skip(t.Name() + "requires 'samlUser' and 'samlPassword' to be specified in configuration")
+		return
+	}
+	// Backup default authentication configuration
+	backupUser := testConfig.Provider.User
+	backupPassword := testConfig.Provider.Password
+	backupUseSamlAdfs := testConfig.Provider.UseSamlAdfs
+	backupCustomRptId := testConfig.Provider.CustomAdfsRptId
+	backupSysOrg := testConfig.Provider.SysOrg
+	backupEnableConnectionCache := enableConnectionCache
+	enableConnectionCache = false
+	cachedVCDClients.reset()
+
+	t.Logf("Clearing connection cache and switching credentials to: user=%s, use_saml_adfs=%t, rpt_id=%s, org=%s\n",
+		testConfig.Provider.SamlUser, true, testConfig.Provider.SamlCustomRptId, testConfig.VCD.Org)
+
+	// Temporarily override user and password variables
+	_ = os.Setenv("VCD_USER", testConfig.Provider.SamlUser)
```

Comment:
What happens if `VCD_TOKEN` or `testConfig.Provider.Token` is set? Wouldn't that override the SAML authentication?

Reply:
Well caught! I missed this variable. Changed.